packages/updating-element/src/platform-support.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+/**
+ * UpdatingElement patch to support browsers without native web components.
+ *
+ * @packageDocumentation
+ */
+
+const needsPlatformSupport = !!(
+  window.ShadyCSS !== undefined &&
+  (!window.ShadyCSS.nativeShadow || window.ShadyCSS.ApplyShim)
+);
+
+interface RenderOptions {
+  readonly renderBefore?: ChildNode | null;
+  scope?: string;
+}
+
+const SCOPED = '__scoped';
+
+type CSSResults = Array<{cssText: string} | CSSStyleSheet>;
+
+interface PatchableUpdatingElementConstructor {
+  [SCOPED]: boolean;
+  elementStyles: CSSResults;
+  shadowRootOptions: ShadowRootInit;
+  _handlesPrepareStyles?: boolean;
+}
+
+interface PatchableUpdatingElement extends HTMLElement {
+  new (...args: any[]): PatchableUpdatingElement;
+  constructor: PatchableUpdatingElementConstructor;
+  connectedCallback(): void;
+  hasUpdated: boolean;
+  _afterUpdate(changedProperties: unknown): void;
+  createRenderRoot(): Element | ShadowRoot;
+  _renderOptions: RenderOptions;
+}
+
+(globalThis as any)['updatingElementPlatformSupport'] ??= ({
```

Comment:
I wonder if we need to key the platform support by version. In the case where a newer version of the file is loaded after an older version, we'd want the newer to overwrite so if it contains additional patches they'll take effect.

Reply:
The idea is that platform-support should be loaded once by the application. Loading multiple versions would generally be bad. It's also first one wins, so even if some bad component includes an old platform-support, this is easy to address by loading a new one right when you load your polyfills.

Since this is a corner case, suggest we consider any issues with this as a follow on.